### PR TITLE
Get strings

### PR DESCRIPTION
**Pull request by reeceshuttle**
This pull request implements functions to retrieve the init and goal strings of a PDDL task and refactors code to incorporate the new functions.

### CODE REVIEW

File: llm4pddl/utils.py [hunk 1]
```diff
@@ -384,3 +384,23 @@ def str_to_identifier(x: str) -> str:
         https://stackoverflow.com/questions/5297448
     """
     return hashlib.md5(x.encode('utf-8')).hexdigest()
+
+
+def get_init_str(task: Task) -> str:
+    """Returns the init string of a PDDL task."""
+    domain, problem = parse_task(task)
```
Comment (tomsilver) [inline llm4pddl/utils.py:391]: you can do `_, problem = parse_task(task)` to avoid the `del domain` below

File: llm4pddl/utils.py [hunk 2]
```diff
@@ -384,3 +384,23 @@ def str_to_identifier(x: str) -> str:
...
+    del domain
+    # Create the init string.
+    init_strs = [pred_to_str(p) for p in problem.initial_state]
+    init_str = " ".join(init_strs)
+    return init_str
+
+
+def get_goal_str(task: Task) -> str:
+    """Returns the goal string of a PDDL task."""
+    domain, problem = parse_task(task)
```
Comment (tomsilver) [inline llm4pddl/utils.py:401]: same as above

File: tests/test_utils.py [hunk 3]
```diff
@@ -450,3 +450,35 @@ def test_pyperplan_problem_to_str(domain_file, problem_file):
...
     assert plan is not None
     assert utils.validate_plan(task, plan)
+
+
+def test_get_init_str():
+    """Tests for get_init_str()."""
+    task01 = utils.get_task_from_dir(utils.PYPERPLAN_BENCHMARK_DIR / 'blocks',
+                                     1)
+    init_str = utils.get_init_str(task01)
+    #     assert init_str == "CLEAR(C) CLEAR(A) CLEAR(B) CLEAR(D) ONTABLE(C) \
```
Comment (tomsilver) [inline tests/test_utils.py:460]: remove

File: tests/test_utils.py [hunk 4]
```diff
@@ -450,3 +450,35 @@ def test_pyperplan_problem_to_str(domain_file, problem_file):
...
+    # ONTABLE(A) ONTABLE(B) ONTABLE(D) HANDEMPTY()".lower()
+    assert init_str == "(CLEAR C) (CLEAR A) (CLEAR B) (CLEAR D) (ONTABLE C) \
+(ONTABLE A) (ONTABLE B) (ONTABLE D) (HANDEMPTY)".lower()
+    task02 = utils.get_task_from_dir(utils.PYPERPLAN_BENCHMARK_DIR / 'blocks',
+                                     2)
+    init_str2 = utils.get_init_str(task02)
+    #     assert init_str2 == "CLEAR(B) ONTABLE(D) ON(B C) \
```
Comment (tomsilver) [inline tests/test_utils.py:467]: remove

File: tests/test_utils.py [hunk 5]
```diff
@@ -450,3 +450,35 @@ def test_pyperplan_problem_to_str(domain_file, problem_file):
...
+    assert init_str2 == "(CLEAR B) (ONTABLE D) (ON B C) \
+(ON C A) (ON A D) (HANDEMPTY)".lower()
+
+
+def test_get_goal_str():
+    """Tests for get_goal_str()"""
+    task01 = utils.get_task_from_dir(utils.PYPERPLAN_BENCHMARK_DIR / 'blocks',
+                                     1)
+    goal_str = utils.get_goal_str(task01)
+    # assert goal_str == "on(d c) on(c b) on(b a)"
```
Comment (tomsilver) [inline tests/test_utils.py:478]: remove

File: tests/test_utils.py [hunk 6]
```diff
@@ -450,3 +450,35 @@ def test_pyperplan_problem_to_str(domain_file, problem_file):
...
+    assert goal_str == "(on d c) (on c b) (on b a)"
+    task02 = utils.get_task_from_dir(utils.PYPERPLAN_BENCHMARK_DIR / 'blocks',
+                                     2)
+    goal_str2 = utils.get_goal_str(task02)
+    # assert goal_str2 == "ON(D C) ON(C A) ON(A B)".lower()
```
Comment (tomsilver) [inline tests/test_utils.py:483]: remove

**Review by tomsilver — APPROVED** [review]
excellent!! just a few tiny things

**Comment by reeceshuttle** [comment]
done!

**Pull request merged**